### PR TITLE
Update for Frequency 1.4.0 and DSNP 1.2.0

### PR DESCRIPTION
**Pull request by wilwade**
Problem
=======

Needed to add the schemas for the upcoming DSNP 1.2.0 release.

Change summary:
---------------
* Updated the DSNP schemas for 1.2.0
* Updated to Frequency 1.5.1
* Cleaned up the deploy and read logic and made it easier to add new DSNP Schemas

Steps to Verify:
----------------
1. Run Frequency 1.4.0+ locally
1. `npm run deploy`
1. `npm run read`
1. See that they all deployed and are working!

### CODE REVIEW

File: deploy.ts [hunk 1]
```diff
@@ -75,55 +55,30 @@ const createSchemas = async (schema_names: string[]) => {
...
-    const json = JSON.stringify(schema);
+    const json = JSON.stringify(schemaDeploy?.model);
     // Remove whitespace in the JSON
     const json_no_ws = JSON.stringify(JSON.parse(json));
 
-    let promise;
-
-    // The default model type/payload type is Parquet/IPFS
-    // unless it is a graphChange schema which is AvroBinary/OnChain.
-    if (schemaName === "graphChange") {
```
Comment (wilwade) [inline deploy.ts:88]: Moved the payload location into `dsnpSchemas`

File: dsnp/index.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+type ModelType = "AvroBinary" | "Parquet";
+
+export type Deploy = {
+  model: ParquetModel | object,
+  modelType: ModelType,
+  payloadLocation: PayloadLocation,
+}
+
+// Map schema names (string) to deploy object
+export const dsnpSchemas = new Map<string, Deploy>([
```
Comment (wilwade) [inline dsnp/index.ts:25]: Moved from the deploy script.

File: package.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
   },
   "homepage": "https://github.com/LibertyDSNP/schemas#readme",
   "dependencies": {
-    "@frequency-chain/api-augment": "1.2.0",
+    "@frequency-chain/api-augment": "1.4.0-rc3",
```
Comment (wilwade) [inline package.json:25]: TODO: Update to 1.4.0 once it is released.

File: read.ts [hunk 4]
```diff
@@ -3,23 +3,9 @@ import { getEndpoint, getFrequencyAPI } from "./services/connect";
...
-const nameAndSchema = [
-  ["dsnp.broadcast", JSON.stringify(broadcast)],
-  ["dsnp.profile", JSON.stringify(profile)],
-  ["dsnp.reaction", JSON.stringify(reaction)],
-  ["dsnp.reply", JSON.stringify(reply)],
-  ["dsnp.tombstone", JSON.stringify(tombstone)],
-  ["dsnp.update", JSON.stringify(update)],
-  ["dsnp.graphChange", JSON.stringify(graphChange)],
-];
+const nameAndSchema = Array.from(dsnpSchemas.entries(), ([k, v]) => ([`dsnp.${k}`, JSON.stringify(v.model)]));
```
Comment (wilwade) [inline read.ts:8]: I always forget that Array.from can map, but it can.

File: dsnp/userPrivateFollows.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
+// Paginated Chunk with PRIds
```
Comment (JoeCap08055) [inline dsnp/userPrivateFollows.ts:1]: Remove "with PRIds" if we're dropping them

**Comment by JoeCap08055** [comment]
I see we're adding a "Private Follows" schema, which is similar to "Private Connections", but without prids. Are we sure about that? I realize we don't need PRIds for messaging with follows, but wouldn't they still be useful for proof-of-follow?

**Comment by wilwade** [comment]
@JoeCap08055 I think we need to keep them out for now. The DSNP Spec doesn't specify it and I think proof of following needs to be investigated separately.

**Review by JoeCap08055 — APPROVED** [review]
lgtm

**Review by shannonwells — APPROVED** [review]
LGTM & WOMM
(looks good to me and works on my machine)

**Pull request merged**